### PR TITLE
tests(@qml): add QML tests for onboarding scenarios + pinned messages modal

### LINKED ISSUE #22196: Replace onboarding scenarios with QML equivalents
A couple of tests could be moved from e2e to QML level without coverage and quality loss

### PR DESCRIPTION
**Pull request by anastasiyaig**
### What does the PR do

Fix https://github.com/status-im/status-app/issues/22196

### CODE REVIEW

File: ui/app/AppLayouts/Chat/popups/PinnedMessagesPopup.qml [hunk 1]
```diff
@@ -61,6 +63,16 @@ StatusDialog {
...
 
+        readonly property var chatContentModule: QtObject {
+            readonly property var chatDetails: QtObject {
+                readonly property string id: root.chatId
+                readonly property int type: Constants.chatType.oneToOne
+                readonly property bool canPostReactions: false
+                readonly property bool canPost: true
+                readonly property bool canView: true
+            }
+        }
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/popups/PinnedMessagesPopup.qml]: I don't think this belongs here :)

**Review by caybro — COMMENTED** [review]
The rest looks ok 👍

File: ui/app/AppLayouts/Chat/popups/PinnedMessagesPopup.qml (hunk 1 above)
Comment (anastasiyaig) [reply ui/app/AppLayouts/Chat/popups/PinnedMessagesPopup.qml]: removed! thanks

Comment (caybro) [reply ui/app/AppLayouts/Chat/popups/PinnedMessagesPopup.qml]: No, I meant the whole thing; why do you need this nested structure at all?

Comment (anastasiyaig) [reply ui/app/AppLayouts/Chat/popups/PinnedMessagesPopup.qml]: i was trying to get rid of `TypeError: Cannot read property 'chatDetails' of undefined` when the tests for popup instantiated it without real chatContentModule

all removed now, sorry about that

Comment (caybro) [reply ui/app/AppLayouts/Chat/popups/PinnedMessagesPopup.qml]: No problem, I thought it was masking some other issue, and not really needed :)

File: storybook/pages/PinnedMessagesPopupPage.qml [hunk 2]
```diff
@@ -274,15 +274,6 @@ SplitView {
...
 
-                property var chatContentModule: QtObject {
-                    property var chatDetails: QtObject {
-                        property bool canPostReactions: true
-                        property bool canPost: true
-                        property bool canView: true
-                    }
-                    property var pinnedMessagesModel: mockMessageStore.pinnedMessagesModel
-                }
-
```
Comment (caybro) [inline storybook/pages/PinnedMessagesPopupPage.qml:285]: This otoh is useful here :) It emulates the backend structure

File: storybook/qmlTests/tests/tst_PasswordView.qml [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+            verify(!!newInput)
+            verify(!!confirmInput)
+
+            newInput.text = "short"
+            confirmInput.text = "short"
+            compare(confirmButton.enabled, false)
+
+            newInput.text = "ValidPass1!"
+            confirmInput.text = "ValidPass1!"
+            tryCompare(confirmButton, "enabled", true)
```
Comment (caybro) [inline storybook/qmlTests/tests/tst_PasswordView.qml:210]: Try also testing empty passwords, and passwords containing Unicode characters

**Review by caybro — COMMENTED** [review]
LGTM in general

File: storybook/qmlTests/tests/tst_PasswordView.qml (hunk 3 above)
Comment (anastasiyaig) [reply storybook/qmlTests/tests/tst_PasswordView.qml:210]: good point, added

**Review by caybro — APPROVED** [review]
Nice 👍
